include/adm/utilities/id_assignment.hpp

```diff
@@ -93,4 +97,48 @@ namespace adm {
   }
   ///@}
 
+  class IdReassigner {
```

Comment:
Could this go in the implementation file? It doesn't look like it should be used by users.

It might be worth a quick comment on IdReassigner and IdIssuer to say what they are for (especially in the presence if IdAssigner, which could also do with explaining itself).

I'm fine with the structure as it is, but I would be tempted to merge these into one class, and perhaps pass the document to the `reassignXIds` functions instead of storing it in the class.

Reply:
I think passing document to IdReassigner::reassignAllIds might suggest that a single IdReassigner instance can be used to process multiple documents, which is fine, but it would mean reconstructing IdIssuer on each passed in document (also fine). Since that IdIssuer instance is associated with just that one document, they should be passed to functions together. I think then the code would become less clean. That said, I have no strong feeling either way...

Reply:
I guess we could wrap a Document with its IdIssuer in a struct and pass that around

Reply:
> I think passing document to IdReassigner::reassignAllIds might suggest that a single IdReassigner instance can be used to process multiple documents

Oh yeah, true, it looks good to me as-is.